### ISSUE TITLE
No call was registered or all methods took <1ms

### ISSUE THREAD

**Issue by jbeckers** [kornilova-l/flamegraph-visualizer#4]
What should I try when all results I ever get are "No call was registered or all methods took <1ms" instead of a flame graph?

This is the beginning of the log output in the IJ console:
> "C:\Program Files (x86)\JWS\JavaSoft\JRE\jre1.8.0_131\bin\java" -Djava.library.path=C:\System9\KWS\lib\dll -Xms512m -Xmx512m -XX:+HeapDumpOnOutOfMemoryError -ea -Ddefaultprinter=PDFCreator -Ddefaultserver=PROD_S9 -Dhttps.protocols=TLSv1,SSLv2Hello -Dicon=/uz/s9client/kws/images/kws_dyn.gif -Dlocal_actions=sourcesEJB\uz\s9server\services\toegang\driver -Dlocal_settings=sourcesEJB\uz\s9server\services\toegang\driver -Dlog4j.configuration=uz/s9/kws/log4j.properties -Dmultidesktop.create.default.on.primary.screen=true -Drebel.packages=uz -DS9QueryLogic.debug.logNonUZCode=true -DS9QueryLogic.debug.multiline=true -DS9QueryLogic.debug.nbStackInfo=10,10 -DS9QueryLogic.debug.outOfOrder=true -DS9QueryLogic.debug.suppressGetAttributes=false -DS9QueryLogic.debug.threadinfo=true -DS9QueryLogic.debug=true "-Dtitle=Dynamisch KWS (preprod lokaal) (jre8) (MultiDesktop)" -Dwlpreprod=true -DJINTEGRA_NATIVE_MODE=true -Dsun.java2d.noddraw=true "-javaagent:C:\uz\Settings\IntelliJIdea\config\plugins\FlamegraphProfiler\classes\javaagent.jar=C:\uz\Settings\IntelliJIdea\system\flamegraph-profiler\log\KWS\KWS preprod local 8 MultiDesktop-2018-01-22-15_21_52.ser&C:\uz\Settings\IntelliJIdea\system\flamegraph-profiler\configuration\KWS.config" -javaagent:C:\uz\ProgramFiles\IntelliJIDEA2017.3EAP\lib\idea_rt.jar=64468:C:\uz\ProgramFiles\IntelliJIDEA2017.3EAP\bin -Dfile.encoding=windows-1252 -classpath C:\Users\jbecke4\AppData\Local\Temp\classpath1.jar uz.s9client.kws.Kws
> Output file: C:\uz\Settings\IntelliJIdea\system\flamegraph-profiler\log\KWS\KWS preprod local 8 MultiDesktop-2018-01-22-15_21_52.ser
> Configuration:
> Including patterns: 
>	*.*(*+)+

Can I find/add any other debug log info?

**Comment by kornilova-l** [comment]
@jbeckers, When your program finishes you can see how many method calls were registered.
It looks like this:
```
Output file: /home/lk/.IntelliJIdea2017.3/system/flamegraph-profiler/log/untitled/Main-2018-01-23-23_06_13.ser
Configuration:
Including patterns: 
	*.*(*)
Methods count: 145

Process finished with exit code 0
```
If it is `Methods count: 0` then there are two possible situations:
1. No method was instrumented because configuration is incorrect
2. All methods were quick and therefore were not registered (you do not want to use instrumentation profiler to collect info about small methods because in this case instrumentation adds visible overhead)

I think that in your case you have problem with configuration which is `.(*)` The configuration is strange and I could not reproduce this result. It seems like it has empty class pattern and empty method pattern. my IDEA fields validation does not allow this.
Try to edit configuration so  class pattern and method pattern is `*`. It means that your configuration will look like this: `*.*(*+)+`

I personally often use Java Flight Recorder to collect info and then upload file to the visualizer. Try to run program with following VM options: `-XX:+UnlockCommercialFeatures -XX:+FlightRecorder -XX:StartFlightRecording=duration=10s,filename=my_recording.jfr` and then open my_recording.jfr in visualizer. You may find this option more convenient and suitable for your needs.

If something is still unclear or it does not work then add comment to this issue. If there will be no comments I will close the issue.

**Comment by jbeckers** [comment]
Ok, thanks for the explanation. I do feel like a beginner wrt flamegraphs, this is valuable info.

Some findings so far:
* The configuration I used is the one you suggest. I think github must have scrambled it in the quote. I've now double checked and it's `*.*(*+)+`
* I did not realize the results were only available after the program ended. This differs from other profilers.
* When I do wait until the end of the program, I see this:
> Methods count: 0
>  JNIutil::init : 28E4FAA8
> max user process virtual address = 7FFEFFFF
* There might be an issue where this plugin interferes with the new debugging profiler included in IDEA. I'll try to disable that one.
* Thanks for the jfr suggestion. Also on my list of things to try.
